pkg/cli/environment.go

```diff
@@ -53,9 +55,20 @@ type EnvSettings struct {
 	PluginsDirectory string
 }
 
+var (
+	config     genericclioptions.RESTClientGetter
+	configOnce sync.Once
+
+	// Namespace is the namespace scope.
+	namespace string
+	// KubeConfig is the path to the kubeconfig file.
+	kubeConfig string
```

Comment:
Same here RE: making this a shared global variable. This should be part of the EnvSettings object as it was before, but as an internal variable.